hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -437,6 +451,18 @@ private Table newHmsTable(TableMetadata metadata) {
             null,
             TableType.EXTERNAL_TABLE.toString());
 
+    // hive2 does not support setting table ownership type, therefore we leave it as
+    // a table property. Once upgraded to hive3, then we can simply set
+    // newTable.setOwnerType(PrincipalType.valueOf(metadata.get(HMS_TABLE_OWNER_TYPE)))
+    String ownerType = metadata.property(TableProperties.HMS_TABLE_OWNER_TYPE, null);
+    if (ownerType != null && ownerType.equalsIgnoreCase(PrincipalType.GROUP.name())) {
+      newTable
+          .getParameters()
+          .put(TableProperties.HMS_TABLE_OWNER_TYPE, PrincipalType.GROUP.name());
+    } else { // default: individual ownership
```

Comment:
I think instead of this else branch you could pass PrincipalType.USER.name() to metadata.property() instead of null as a second parameter in L457 that is in fact the default value when the first param is not a valid key.

Reply:
Good idea.